code/controllers/subsystem/processing/greyscale.dm

```diff
@@ -27,12 +27,68 @@ PROCESSING_SUBSYSTEM_DEF(greyscale)
 		var/datum/greyscale_config/config = configurations[greyscale_type]
 		config.CrossVerify()
 
+	if(CONFIG_GET(flag/generate_assets_in_init))
+		ExportMapPreviews()
+
 	return SS_INIT_SUCCESS
 
 /datum/controller/subsystem/processing/greyscale/proc/RefreshConfigsFromFile()
 	for(var/i in configurations)
 		configurations[i].Refresh(TRUE)
 
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviews()
```

Comment:
Do you have profiles for this?

Reply:
No but why does it matter, it only runs in init on dev machines. I do know it's less than 6 seconds, just from looking at subsystem init time.

Reply:
I care about init time. This isn't a blocker but I want to be conscious of it and profiles help me do that.

Reply:
I'll grab the profiler data from the proc next time I get the chance

Reply:
![image](https://github.com/user-attachments/assets/9dcff882-f7c8-41fc-8e63-ce5b850bb17a)

Reply:
can you send me an export of these? I'd like to look at where that time goes

Reply:
[Profile results (total time).txt](https://github.com/user-attachments/files/16997034/Profile.results.total.time.txt)

Already sent elsewhere, but here it is for anyone else curious. Results are from clearing the profiler first before running the map icon generation.